src/v/cloud_storage/remote_segment.h

```diff
@@ -293,6 +292,19 @@ class remote_segment final {
 
     std::optional<segment_chunks> _chunks_api;
     std::optional<offset_index::coarse_index_t> _coarse_index;
+
+    class consume_stream {
```

Comment:
nit: maybe this can be just a friend class declaration and the whole declaration can go in remote_segment.cc

Reply:
changed